### ISSUE TITLE
Option to adjust lights linearly vs log

### ISSUE THREAD

**Issue by BaelfireNightshd** [iRayanKhan/homebridge-tuya#88]
My lights (Feit Electric RGBTW model: OM60/RGBW/CA/AG/4 ?) don’t adjust linearly. So 50% is barely noticeable and I really have to get down below 25% to nice dimming. It would be nice if there was a flag I could set to switch between linear and logarithmic dimming for different lights. 

I’ll see about doing research on dimming curves. But it looks like we’re looking for a logarithmic dimming curve (and doesn’t look like anyone wants to give up the formula for the curve they use)

**Comment by BaelfireNightshd** [comment]
After doing some research, I came across http://pw.boulderis.com/blogposts/details/44

Which states that the perceived amount is about sqrt(percentAsADecimal). So for example for 10% brightness, you’d see it as sqrt(0.1) = 31.6% perceived brightness. I also noticed that some papers said that at 50% brightness you get about 80% perceived brightness. Which is actually more in line with a cube root. What could be done is have a variable that’s settable in the config and just do the follow calculation

`outputBrightness = maxBrightness * ((input brightness / maxBrightness) ^ (x))`

Where x is the variable set in config.json. 

If x=1 that’s the same as linear, 2 is square, and 3 is cubed. 

Note that I’m taking the square to cancel out the square root (or cube).
